devdoc/constbuffer_requirements.md

```diff
@@ -116,6 +118,30 @@ The memory has to be free by calling the custom free function passed as argument
 
 **SRS_CONSTBUFFER_01_011: [** If any error occurs, `CONSTBUFFER_CreateWithMoveMemory` shall fail and return NULL. **]**
 
+### CONSTBUFFER_CreateFromOffsetAndSize
+```c 
+MOCKABLE_FUNCTION(, CONSTBUFFER_HANDLE, CONSTBUFFER_CreateFromOffsetAndSize, CONSTBUFFER_HANDLE, handle, size_t, offset, size_t, size)
+```
+
+Given an existing `handle` `CONSTBUFFER_CreateFromOffsetAndSize` creates another `CONSTBUFFER_HANDLE` from `size` bytes  `handle` starting at `offset`.
+
+**SRS_CONSTBUFFER_02_025: [** If `handle` is `NULL` then `CONSTBUFFER_CreateFromOffsetAndSize` shall fail and return `NULL`. **]**
+
+**SRS_CONSTBUFFER_02_026: [** If `offset` is greater than or equal to `handles`'s size then `CONSTBUFFER_CreateFromOffsetAndSize` shall fail and return `NULL`. **]**
+
+**SRS_CONSTBUFFER_02_027: [** If `offset` + `size` exceed `handles`'s size then `CONSTBUFFER_CreateFromOffsetAndSize` shall fail and return `NULL`. **]**
+
+**SRS_CONSTBUFFER_02_028: [** `CONSTBUFFER_CreateFromOffsetAndSize` shall allocate memory for a new `CONSTBUFFER_HANDLE`'s content. **]**
+
+**SRS_CONSTBUFFER_02_029: [** `CONSTBUFFER_CreateFromOffsetAndSize` shall set the ref count of the newly created `CONSTBUFFER_HANDLE` to the initial value. **]**
+
```

Comment:
Consider having the memory for the new buffer be the memory of the old buffer.
As in: inc_ref the original buffer, set the free function to do a dec ref of the original buffer.

Just saying might be interesting to take advantage of the fact that this is already all ref counted.

Reply:
Oh looked in the code and that is what we do

---
In reply to: [284970951](https://github.com/Azure/azure-c-shared-utility/pull/321#discussion_r284970951) [](ancestors = 284970951)

Reply:
Noice!

---
In reply to: [284971097](https://github.com/Azure/azure-c-shared-utility/pull/321#discussion_r284971097) [](ancestors = 284971097,284970951)